tests/components/lcn/test_switch.py

```diff
@@ -0,0 +1,254 @@
+    """Test the relay switch turns off."""
+    states = [RelayStateModifier.NOCHANGE] * 8
+    states[0] = RelayStateModifier.OFF
+
+    state = hass.states.get("switch.switch_relay1")
+    state.state = STATE_ON
+
+    # command failed
+    control_relays.return_value = False
+
+    await hass.services.async_call(
+        DOMAIN_SWITCH,
+        SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: "switch.switch_relay1"},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    control_relays.assert_awaited_with(states)
+
+    state = hass.states.get("switch.switch_relay1")
+    assert state.state == STATE_ON
+
+    # command success
+    control_relays.reset_mock(return_value=True)
+    control_relays.return_value = True
+
+    await hass.services.async_call(
+        DOMAIN_SWITCH,
+        SERVICE_TURN_OFF,
+        {ATTR_ENTITY_ID: "switch.switch_relay1"},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+    control_relays.assert_awaited_with(states)
+
+    state = hass.states.get("switch.switch_relay1")
+    assert state.state == STATE_OFF
+
+
+async def test_pushed_output_status_change(hass, entry, lcn_connection):
+    """Test the output switch changes its state on status received."""
+    device_connection = get_device_connection(hass, (0, 7, False), entry)
+    address = LcnAddr(0, 7, False)
+
+    # push status "on"
+    inp = ModStatusOutput(address, 0, 100)
+    await device_connection.async_process_input(inp)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("switch.switch_output1")
+    assert state.state == STATE_ON
+
+    # push status "off"
+    inp = ModStatusOutput(address, 0, 0)
+    await device_connection.async_process_input(inp)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("switch.switch_output1")
+    assert state.state == STATE_OFF
+
+
+async def test_pushed_relay_status_change(hass, entry, lcn_connection):
+    """Test the relay switch changes its state on status received."""
+    device_connection = get_device_connection(hass, (0, 7, False), entry)
+    address = LcnAddr(0, 7, False)
+    states = [False] * 8
+
+    # push status "on"
+    states[0] = True
+    inp = ModStatusRelays(address, states)
+    await device_connection.async_process_input(inp)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("switch.switch_relay1")
+    assert state.state == STATE_ON
+
+    # push status "off"
+    states[0] = False
+    inp = ModStatusRelays(address, states)
+    await device_connection.async_process_input(inp)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("switch.switch_relay1")
+    assert state.state == STATE_OFF
+
+
+async def test_unload_config_entry(hass, entry, lcn_connection):
+    """Test the switch is removed when the config entry is unloaded."""
+    await hass.config_entries.async_forward_entry_unload(entry, DOMAIN_SWITCH)
```

Comment:
Please unload the whole config entry instead of platform only. `async_forward_entry_unload` isn't expected to be called externally from outside the intergration

Reply:
Replaced by `async_unload`.